src/ompl/util/RandomNumbers.h

```diff
@@ -139,6 +146,30 @@ namespace ompl
             return localSeed_;
         }
 
+        /** \brief Uniform random sampling of a unit-length vector. I.e., the surface of an n-ball */
+        void uniformNormalVector(unsigned int n, double value[]);
+
+        /** \brief Uniform random sampling of the content of an n-ball, with a radius appropriately distributed between [0,r) so that the distribution is uniform in a Cartesian coordinate system. */
+        void uniformInBall(double r, unsigned int n, double value[]);
+
+        /** \brief Uniform random sampling of the surface of a prolate hyperspheroid, a special symmetric type of
+        n-dimensional ellipse.
+        @par J D. Gammell, S. S. Srinivasa, T. D. Barfoot, "Informed RRT*: Optimal Sampling-based
+        Path Planning Focused via Direct Sampling of an Admissible Ellipsoidal Heuristic."
+        IROS 2014. DOI: <a href="http://dx.doi.org/10.1109/IROS.2014.6942976">10.1109/IROS.2014.6942976</a>.
+        <a href="http://www.youtube.com/watch?v=d7dX5MvDYTc">Illustration video</a>.
+        <a href="http://www.youtube.com/watch?v=nsl-5MZfwu4">Short description video</a>. */
+        void uniformProlateHyperspheroidSurface(ProlateHyperspheroidPtr phsPtr, unsigned int n, double value[]);
```

Comment:
Please pass the ptr as const &, or probably better, const & to the class itself. As it is, it makes a copy & then destroys that copy of a shared ptr with every call. Due to the atomic increment / decrement, efficiency is affected.